projects/netgrif-components/src/lib/navigation/navigation-double-drawer/navigation-double-drawer.component.html

```diff
@@ -157,7 +157,7 @@
              [routerLink]="imageRouterLink !== undefined ? imageRouterLink : []">
     </div>
     <div *ngIf="(rightLoading$ | async) === false" ngClass.gt-sm="padding-top-8">
-        <div class="margin-auto cursor-pointer" *ngIf='!isRightNodesEmpty()' (click)="toggleSection('folders')"
+        <div class="margin-auto cursor-pointer" *ngIf='!(rightNodes | isEmpty)' (click)="toggleSection('folders')"
```

Comment:
how does the `!(input | pipe)` syntax perform in terms of performance? Is the negation applied only if the pipe changes? Would it not be better to implement a `isNonEmpty` pipe, that contains the negation?

Reply:
As Angular pipe is only a pure function the performance hit by negation depends on the pipe implementation. In this case, it is irrelevant as every time the pipe resolves into a boolean so negotiation is not taxing on performance. Nevertheless, I implemented the `inNonEmpty` pipe. It is more elegant 😄